src/table/headerRenderers/dataHeader.component.tsx

```diff
@@ -84,58 +92,82 @@ const DataHeader = (props: DataHeaderProps): React.ReactElement => {
   );
 
   return (
-    <TableCell
-      size="small"
-      component="th"
-      role="columnheader"
-      sx={sx}
-      variant="head"
-      sortDirection={currSortDirection}
+    <Draggable
+      draggableId={dataKey}
+      index={index}
+      isDragDisabled={!permitResizing}
     >
-      <div
-        style={{
-          overflow: 'hidden',
-          flex: 1,
-        }}
-      >
-        <Box
-          aria-label={`${dataKey} header`}
-          display="flex"
-          onMouseDown={(event) => {
-            // Middle mouse button can also fire onClose
-            if (event.button === 1) {
-              event.preventDefault();
-              onClose(dataKey);
-            }
-          }}
-        >
-          <Box marginRight={1}>{Icon && <Icon />}</Box>
-          <Box>{inner}</Box>
-          <div aria-label={`close ${dataKey}`}>
-            <StyledClose onClick={() => onClose(dataKey)} />
-          </div>
-        </Box>
-      </div>
-      {/* Draggable? */}
-      <div>
-        <div
-          style={{
-            marginLeft: 18,
-            paddingLeft: '4px',
-            paddingRight: '4px',
-            cursor: 'col-resize',
-          }}
-        >
-          <Divider
-            orientation="vertical"
-            flexItem
-            sx={{
-              height: '100%',
-            }}
-          />
-        </div>
-      </div>
-    </TableCell>
+      {(provided) => {
+        return (
+          <TableCell
+            {...provided.draggableProps}
+            ref={provided.innerRef}
+            {...provided.dragHandleProps}
+            size="small"
+            component="th"
+            role="columnheader"
+            sx={sx}
+            variant="head"
+            sortDirection={currSortDirection}
+          >
+            <div
+              style={{
+                overflow: 'hidden',
+                flex: 1,
+              }}
+            >
+              <Box
+                aria-label={`${dataKey} header`}
+                display="flex"
+                onMouseDown={(event) => {
+                  // Middle mouse button can also fire onClose
+                  if (event.button === 1) {
+                    event.preventDefault();
+                    onClose(dataKey);
+                  }
+                }}
+              >
+                <Box marginRight={1}>{Icon && <Icon />}</Box>
+                <Box>{inner}</Box>
+              </Box>
+            </div>
+            <div
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+              }}
+              onMouseOver={() => {
+                setPermitResizing(false);
+              }}
+              onMouseOut={() => {
+                setPermitResizing(true);
+              }}
```

Comment:
Most of the loss of codecov here is over these two events. Any advice on how to test these (or even if we should) is appreciated.